src/garage/tf/models/cnn_mlp_merge_model.py

```diff
@@ -0,0 +1,141 @@
+            None to maintain a linear activation.
+        hidden_w_init (callable): Initializer function for the weight
+            of intermediate dense layer(s) in the MLP. The function should
+            return a tf.Tensor.
+        hidden_b_init (callable): Initializer function for the bias
+            of intermediate dense layer(s) in the MLP. The function should
+            return a tf.Tensor.
+        output_nonlinearity (callable): Activation function for output dense
+            layer in the MLP. It should return a tf.Tensor. Set it to None
+            to maintain a linear activation.
+        output_w_init (callable): Initializer function for the weight
+            of output dense layer(s) in the MLP. The function should return
+            a tf.Tensor.
+        output_b_init (callable): Initializer function for the bias
+            of output dense layer(s) in the MLP. The function should return
+            a tf.Tensor.
+        layer_normalization (bool): Bool for using layer normalization or not.
+    """
+
+    def __init__(self,
+                 filter_dims,
+                 num_filters,
+                 strides,
+                 hidden_sizes=[256],
+                 action_merge_layer=-2,
+                 name=None,
+                 padding='SAME',
+                 max_pooling=False,
+                 pool_strides=(2, 2),
+                 pool_shapes=(2, 2),
+                 cnn_hidden_nonlinearity=tf.nn.relu,
+                 cnn_hidden_w_init=tf.glorot_uniform_initializer(),
+                 cnn_hidden_b_init=tf.zeros_initializer(),
+                 hidden_nonlinearity=tf.nn.relu,
+                 hidden_w_init=tf.glorot_uniform_initializer(),
+                 hidden_b_init=tf.zeros_initializer(),
+                 output_nonlinearity=None,
+                 output_w_init=tf.glorot_uniform_initializer(),
+                 output_b_init=tf.zeros_initializer(),
+                 layer_normalization=False):
+        super().__init__(name)
+
+        if not max_pooling:
+            self.cnn_model = CNNModel(
+                filter_dims=filter_dims,
+                num_filters=num_filters,
+                hidden_w_init=cnn_hidden_w_init,
+                hidden_b_init=cnn_hidden_b_init,
+                strides=strides,
+                padding=padding,
+                hidden_nonlinearity=cnn_hidden_nonlinearity)
+        else:
+            self.cnn_model = CNNModelWithMaxPooling(
+                filter_dims=filter_dims,
+                num_filters=num_filters,
+                hidden_w_init=cnn_hidden_w_init,
+                hidden_b_init=cnn_hidden_b_init,
+                strides=strides,
+                padding=padding,
+                pool_strides=pool_strides,
+                pool_shapes=pool_shapes,
+                hidden_nonlinearity=cnn_hidden_nonlinearity)
+
+        self.mlp_merge_model = MLPMergeModel(
+            output_dim=1,
```

Comment:
It should not be 1, right? It should be `output_dim` specified by user

Reply:
In the case of continuous action spaces , since we only predict one Q value for the given state, action pair we only need 1 scalar output. This is how it's done in `ContinuousMLPQFunction` as well.  `DiscreteCNNQFunction` and `DiscreteMLPQfunction` set `output_dim=action_dim` because they output a Q value for each action (I assume).

Reply:
You are right about ContinuousCNNQFunction but `CNNMLPMergeModel` may be used for other policies as well, so it should be generic enough to be reused. We can then set `output_dim` to 1 in `ContinuousCNNQFunction`.

Reply:
Ah you're right, fixed.